### PR TITLE
fix(scrollable-region-focusable): report inapplicable if scrollable regions are disabled

### PR DESCRIPTION
**Pull request by Zidious**
This PR checks if the scrollable content has the `disabled` attribute, if so we can ignore it as it is not focusable by the keyboard

Closes issue: https://github.com/dequelabs/axe-core/issues/3692, https://github.com/dequelabs/axe-core/issues/3691

### CODE REVIEW

File: lib/rules/scrollable-region-focusable-matches.js [hunk 1]
```diff
@@ -54,6 +55,14 @@ function scrollableRegionFocusableMatches(node, virtualNode) {
     }
   }
 
+  /**
+   * if scrollable content is disabled (not focusable)
+   * we can return false
+   */
+  if (attrNames.includes('disabled')) {
```
Comment (straker) [inline lib/rules/scrollable-region-focusable-matches.js:62]: We should use `commons.forms.isDisabled`. It'll cover things like `disabled` being inherited from a parent (such as `fieldset`), and using `aria-disabled`

File: test/rule-matches/scrollable-region-focusable-matches.js [hunk 2]
```diff
@@ -1,24 +1,60 @@
...
+        '<option value="First">First</option>' +
+        '<option value="Second">Second</option>' +
+        '<option value="Third">Third</option>' +
+        '</select>'
+    );
+    var actual = rule.matches(target.actualNode, target);
+    assert.isFalse(actual);
+  });
+
+  it('returns false when the scrollable content is disabled="true" and not focusable', function () {
```
Comment (straker) [inline test/rule-matches/scrollable-region-focusable-matches.js:33]: Should add a test for parent fieldset having `disabled` as well as `aria-disabled`

**Review by straker — CHANGES_REQUESTED** [review]

File: test/rule-matches/scrollable-region-focusable-matches.js (hunk 2 above)
Comment (Zidious) [reply test/rule-matches/scrollable-region-focusable-matches.js:33]: If the parent element is disabled does that not mean all children are disabled by default?

Comment (Zidious) [reply test/rule-matches/scrollable-region-focusable-matches.js:33]: edit:  added test for this scenario: 

```html
  <fieldset disabled>
    <legend>Do you like receiving wrenches?</legend>
    <input
      type="checkbox"
      id="chbx"
      name="agree"
      value="Yes!"
      aria-disabled="false"
    />
    <label for="chbx">Well, yes of course!</label>
  </fieldset>
  ```

**Comment by Zidious** [comment]
After looking into this for [firefox CI failure](https://app.circleci.com/pipelines/github/dequelabs/axe-core/4339/workflows/632d1756-f16a-4373-b68b-052bc844b17d/jobs/49833?invite=true#step-109-9244). 

With the following snippet: 

```html
<textarea id="pass8" name="textarea" aria-label="test">
  hazaar
  hazaar
  hazaar
</textarea>
```

Within `scrollable-region-focusable` we call `getScroll()` which calls this function [isScrollable()](https://github.com/dequelabs/axe-core/blob/develop/lib/core/utils/get-scroll.js#L37-L40). 

Here we check the `scrollableX` and `scrollableY` values have the `scroll` or `overflow` prop;

With Chrome we get: 

```
{overflowProp: 'auto'}
```

Interestingly, with Firefox we get: 

```
{ overflowProp: "visible" }
```

I was not sure if adding `visible` to the `overflowProp` had some possible negative side effects - possibly requires more discussion 👀

**Pull request closed without merging**